schemas/file.yml

```diff
@@ -25,67 +38,92 @@
   - name: extension
     level: extended
     type: keyword
-    short: File extension.
-    description: >
-      File extension.
-
-      This should allow easy filtering by file extensions.
+    description: File extension.
     example: png
 
   - name: type
     level: extended
     type: keyword
     description: File type (file, dir, or symlink).
+    example: file
 
   - name: device
     level: extended
     type: keyword
     description: Device that is the source of the file.
+    example: sda
+
+  - name: system
```

Comment:
I'm wondering if this belongs here. `file` is information about a specific file. I see how it could fit it in but stumbled over it.

The other part I would not call it `system` as I expect us to have a system object in the future related to system metrics. `system` already exists today in Metricbeat. Alternative names?

Reply:
`file.filesystem`?

Reply:
I'm pretty strongly in favour of adding this field even though it's not information directly about a given file. It's still useful metadata about the place where a file lives. I wanted to point to `host` as very general metadata about an event, but `host` is a field set, not a field inside one specific field set.

Perhaps we could do the same with filesystem information, and introduce a `volume` field set?

Volume fields:

- name
- filesystem
- size (bytes)
- used (bytes)
- letter (Windows drive letter)
- kind (ssd, spinning rust)
- type (platform specific types, such as AWS' gp2, io1 etc)
- id
- serial_number
...